### PR TITLE
add license notice change

### PR DESCRIPTION
**Pull request by adrinjalali**
Following https://github.com/scikit-learn/scikit-learn/issues/20813#issuecomment-2075580619 , this is the public announcement that we'll be changing the license info in the codebase.

### CODE REVIEW

File: _posts/2024-05-04-authorship-info.markdown [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# Authors: Author1, Author2, ...
+# License: BSD 3 clause
+```
+
+However, after a series of discussions which you can see in detail in [this
+issue]( https://github.com/scikit-learn/scikit-learn/pull/28799), could list
+the following caveats to the status quo:
+
+- Authorship information was not up-to-date and in most cases, but not always,
+  reflect the original authors of the file;
```
Comment (lorentzenchr) [inline _posts/2024-05-04-authorship-info.markdown:46]:
```suggestion
  does not reflect the original authors of the file;
```
Did you forget a "not"?

Comment (adrinjalali) [reply _posts/2024-05-04-authorship-info.markdown:46]: no I didn't, I think in most cases people do NOT update the authorship, so in most cases it does reflect the original authors.

**Review by jjerphan — APPROVED** [review]
Thank you, @adrinjalali.

**Review by thomasjpfan — COMMENTED** [review]
Is there a legal backing for doing this? Specifically, is there any precedent from other open source projects that we can use as a guide?

I'm imaging this being shared around on Linkedin or reddit and there will be mixed opinions on this. I do not want this <s>licensing</s> authors change to stop people from contributing.

Edit: I meant "authors".

**Comment by lorentzenchr** [comment]
> licensing change

This is **not** a change of license. We have to be careful with wordings.

**Comment by adrinjalali** [comment]
> Specifically, is there any precedent from other open source projects that we can use as a guide?

We can always revert if an issue comes in.

> I'm imaging this being shared around on Linkedin or reddit and there will be mixed opinions on this. I do not want this licensing author change to stop people from contributing.

I'd say it's the other way around. I remember working on issues and realizing no matter how much I work, my name is not gonna be on top of the file and that was a bit demotivating. We haven't been really adding names on top of the existing files for a long time really, and that means most active contributors are not there and they'd appreciate this change.

**Comment by GaelVaroquaux** [comment]
As someone who's name is on many files and will be removed, I support this change.

These names do not reflect the full history of the files.

The original policy was designed without having in mind the huge growth of the community.

**Comment by agramfort** [comment]
it's fine also for me. Less personal emails from students asking me to look at their code or explain them a function :)

thanks @adrinjalali

**Comment by thomasjpfan** [comment]
> We can always revert if an issue comes in.

If we can revert, then I'm +1 on moving forward with the author change.

File: _posts/2024-05-04-authorship-info.markdown [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+Historically, scikit-learn's files have included authorship information similar
+to the following format:
+
+```python
+# Authors: Author1, Author2, ...
+# License: BSD 3 clause
+```
+
+However, after a series of discussions which you can see in detail in [this
+issue]( https://github.com/scikit-learn/scikit-learn/pull/28799), could list
```
Comment (lorentzenchr) [inline _posts/2024-05-04-authorship-info.markdown:42]: could list? Subject of sentence is missing.

File: _posts/2024-05-04-authorship-info.markdown [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+the following caveats to the status quo:
+
+- Authorship information was not up-to-date and in most cases, but not always,
+  reflect the original authors of the file;
+- It was unfair to all other contributors who have been contributing to the
+  code-base;
+- One can check the real authors and the history of the authors of any part of
+  the code-base using `git blame` and other `git` tools.
+
+Therefore we came to the conclusion to change all inline license information to
```
Comment (lorentzenchr) [inline _posts/2024-05-04-authorship-info.markdown:52]: Do we want to mention the license? If yes, should we focus more on the author part and mention the license info more as a side product (which is not changed, just not always there, maybe with some variations).

Comment (adrinjalali) [reply _posts/2024-05-04-authorship-info.markdown]: Done.

**Comment by adrinjalali** [comment]
Seems like we have enough +1s here, shall we merge?

**Pull request merged**